src/main/java/spoon/support/compiler/SpoonPom.java

```diff
@@ -564,9 +547,6 @@ public static String guessMavenHome() {
 	public String[] buildClassPath(String mvnHome, MavenLauncher.SOURCE_TYPE sourceType, Logger LOGGER, boolean forceRefresh) {
 		if (mvnHome == null) {
 			mvnHome = guessMavenHome();
-			if (mvnHome == null) {
-				throw new SpoonException("M2_HOME must be initialized to use this MavenLauncher constructor.");
-			}
```

Comment:
No longer needed because `guessMavenHome` throws an exception if it does not return a string.